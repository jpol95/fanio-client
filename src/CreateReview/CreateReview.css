.create{
    background-color: lightgray;
    padding: 20px;
}
.create #review{
    width: 75%;
    margin-bottom: 20px;
}