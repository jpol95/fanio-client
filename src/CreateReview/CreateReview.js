import React from "react";
import "./CreateReview.css";
import FanioContext from "../FanioContext";

export default class CreateReview extends React.Component {
  static contextType = FanioContext;
  reviewId = Math.round(Math.random()*1000000000)
  state = {
    id: this.reviewId,
    title: { value: "", touched: false },
    rating: { value: null, touched: false },
    content: { value: "", touched: false },
    tags: { value: [], touched: false },
    errorString: "",
  };

  getTagById = (id) => {
    return this.context.tagList.find(tag => tag.id === Number(id))
  }
  
  displayTagList = () => {
    return this.context.tagList.map((tag) => {
      return (
        <option value={tag.id} key={tag.id} id={tag.id}>
          {tag.name}
        </option>
      );
    });
  };

  handleTitleChange = (title) => {
    const newTitle = { value: title, touched: true };
    this.setState({
      ...this.state,
      title: newTitle,
    });
  };

  handleRatingChange = (rating) => {
    const newRating = { value: rating, touched: true };
    this.setState({
      ...this.state,
      rating: newRating,
    });
  };

  handleTagChange = (tag) => {
    if (this.state.tags.value.includes(tag.value)) return 
    this.setState({
      ...this.state,
      tags: { value: [...this.state.tags.value, tag.value], touched: true },
    });
  };

  handleContentChange = (content) => {
    // console.log(this.state.content)
    const newContent = {value: content, touched: true}
    this.setState({
      ...this.state, content: newContent
    })
  }

  displayTags = () => {
    return this.state.tags.value.map(tag => <span key={tag}>#{this.getTagById(tag).name}</span>)
  }

  handleSubmit = (e) => {
    e.preventDefault()
    const reviewObject = {
      id: this.state.id, 
      title: this.state.title.value, 
      rating: this.state.rating.value,
      content: this.state.content.value, 
      tags: this.state.tags.value, 
    }
    // this.props.history.push(`/users/${}`)
    this.context.handleSubmit(reviewObject)
    
  }


  render() {
    return (
      <form onSubmit={this.handleSubmit} className="create">
        <p>Write a review</p>
        <label htmlFor="title">Title</label>
        <input onChange={(e) => this.handleTitleChange(e.target.value)} id="title" type="text" />
        <label htmlFor="rating">Rating</label>
        <input onChange={(e) => this.handleRatingChange(e.target.value)} id="rating" type="text" />
        <label htmlFor="content">Review</label>
        <textarea onChange={(e) => this.handleContentChange(e.target.value)} id="content" type="text"></textarea>
        <label htmlFor="tags">Tags</label>
        <select  onChange={(e) => this.handleTagChange(e.target)} id="tags" type="text">
        {this.displayTagList()}
        </select>
        <br />
        {this.displayTags()}
        <button>Create Review</button>
      </form>
    );
  }
}
