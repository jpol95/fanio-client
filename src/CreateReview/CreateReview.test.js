import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CreateReview  from './CreateReview';

describe('Create Installment component', () => {
    const props ={
    match: {
        params: {
            userId: 1, 
            installmentId: 1, 
            fandomId: 1, 
            sectionId: 1
        }
    }, 
    installId: 1
}
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CreateReview {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<CreateReview {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});