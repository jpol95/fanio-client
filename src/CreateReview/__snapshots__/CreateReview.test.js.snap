// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Installment component renders the UI as expected 1`] = `
<form
  className="create"
  onSubmit={[Function]}
>
  <label
    htmlFor="title"
  >
    Title
  </label>
  <input
    defaultValue=""
    id="title"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="rating"
  >
    Rating
  </label>
  <input
    defaultValue={null}
    id="rating"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="content"
  >
    Review
  </label>
  <textarea
    defaultValue=""
    id="content"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="tag-input"
  >
    Tags
  </label>
  <input
    id="tag-input"
    list="tags"
    onChange={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Add Tag
  </button>
  <datalist
    id="tags"
    type="text"
  />
  <br />
  <div
    className="tag-list"
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
